code/game/machinery/computer/camera.dm

```diff
@@ -48,51 +78,15 @@ var/global/list/tv_monitors = list()
 	if(!isAI(user))
 		user.set_machine(src)
 
-	var/list/L = list()
-	for (var/obj/machinery/camera/C in cameranet.cameras)
-		L.Add(C)
-
-	camera_sort(L)
-
-	var/list/D = list()
-	D["Cancel"] = "Cancel"
-	for(var/obj/machinery/camera/C in L)
-		if(!istype(C.network, /list))
-			var/turf/T = get_turf(C)
-			WARNING("[C] - Camera at ([T.x],[T.y],[T.z]) has a non list for network, [C.network]")
-			C.network = list(C.network)
-		var/list/tempnetwork = C.network & network
-		if(tempnetwork.len)
-			D[text("[][]", C.c_tag, (C.status ? null : " (Deactivated)"))] = C
-
-	var/t = input(user, "Which camera should you change to?") as null|anything in D
-	if(!t || t == "Cancel")
-		user.cancel_camera()
-		return 0
-	user.set_machine(src)
-
-	var/obj/machinery/camera/C = D[t]
-
-	if(C)
-		if ((!Adjacent(user) || user.machine != src || user.blinded || user.isStunned() || !( C.can_use() )) && (!istype(user, /mob/living/silicon/ai)))
-			if(!C.can_use() && !isAI(user))
-				src.current = null
-			user.cancel_camera()
-			return 0
-		else
-			if(isAI(user))
-				var/mob/living/silicon/ai/A = user
-				A.eyeobj.forceMove(get_turf(C))
-				A.client.eye = A.eyeobj
-			else
-				src.current = C
-				use_power(50)
-
-			spawn(5)
-				attack_hand(user)
-	return
-
-
+	if (!current)
+		init_cams()
+	if (current) // Did we find a camera
```

Comment:
You need to handle the case in which `current.can_use()` is false here. Otherwise using a console that was looking at a previously-usable camera that then became unusuable will render the console useless.